api/v1alpha1/enterprise_types.go

```diff
@@ -3,17 +3,20 @@
 package v1alpha1
 
 import (
+	"github.com/cloudbase/garm/params"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/mercedes-benz/garm-operator/pkg/util/conditions"
 )
 
 // EnterpriseSpec defines the desired state of Enterprise
 type EnterpriseSpec struct {
-	CredentialsName string `json:"credentialsName"`
+	CredentialsRef corev1.TypedLocalObjectReference `json:"credentialsRef"`
```

Comment:
As discussed, let's introduce a new API version (e.g. `v1beta1`) to cover the change in the `CredentialsRef` with a conversion.